src/guides/v2.3/migration/extend-the-tool.md

```diff
@@ -121,7 +152,245 @@ Steps can include four types of classes:
  {:.bs-callout-info}
 Refer to [Configuration]({{ page.baseurl }}/migration/migration-tool-internal-spec.html#configuration), [Step internals]({{ page.baseurl }}/migration/migration-tool-internal-spec.html#step-internals), [Stages]({{ page.baseurl }}/migration/migration-tool-internal-spec.html#step-stages) and [Running modes]({{ page.baseurl }}/migration/migration-tool-internal-spec.html#running-modes) for more information.
 
-Complex SQL queries can be assembled inside these classes to fetch data from the three tables and migrate into a single table. Also, note that these tables should be "ignored" in the [Map Step]({{ page.baseurl }}/migration/migration-tool-internal-spec.html#map-step) because it scans all existing tables and tries to migrate the data unless it is in the `<ignore>` tag of the `map.xml` file.
+Complex SQL queries can be assembled inside these classes to fetch and migrate data. Also, note that these tables should be "ignored" in the [Map Step]({{ page.baseurl }}/migration/migration-tool-internal-spec.html#map-step) because it scans all existing tables and tries to migrate the data unless it is in the `<ignore>` tag of the `map.xml` file.
+
+For Integrity checking we define additional map file in `config.xml`. It will be used to verify that tables structure is like we expect.
+
+```xml
+<config xmlns:xs="http://www.w3.org/2001/XMLSchema-instance"
+        xs:noNamespaceSchemaLocation="urn:magento:module:Magento_DataMigrationTool:etc/config.xsd">
+    ...
+    <options>
+        ...
+        <greatblog_map_file>app/code/Vendor/Migration/etc/opensource-to-opensource/map-greatblog.xml</greatblog_map_file>
+        ...
+    </options>
+</config>
+```
+
+Map file `map-greatblog.xml`:
+
+```xml
+<map xmlns:xs="http://www.w3.org/2001/XMLSchema-instance"
+     xs:noNamespaceSchemaLocation="urn:magento:module:Magento_DataMigrationTool:etc/map.xsd">
+    <source>
+        <field_rules>
+            <ignore>
+                <field>greatblog_post.tags</field>
+            </ignore>
+        </field_rules>
+    </source>
+    <destination>
+        <document_rules>
+            <ignore>
+                <document>greatblog_post_tags</document>
+            </ignore>
+        </document_rules>
+    </destination>
+</map>
+```
+
+Integrity checking class extends `Migration\App\Step\AbstractIntegrity` and contains `perform` method where we verify tables structure:
+
+```php
+class Integrity extends \Migration\App\Step\AbstractIntegrity
```

Comment:
In the following examples, we provide info about namespaces of the newly created classes i.e. `Vendor\Migration\Class\Path`. I would suggest providing such info for the current example as well.

Reply:
Changes have been made. Thank you